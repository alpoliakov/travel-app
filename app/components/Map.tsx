import { GoogleMap, LoadScript, Marker } from '@react-google-maps/api';
import React from 'react';

import mapStyles from '../styles/map-styles';

const containerStyle = {
  width: '100%',
  height: '550px',
};

const options = {
  styles: mapStyles,
  disableDefaultUI: true,
  zoomControl: true,
  fullscreenControl: true,
  streetViewControl: true,
  minZoom: 3,
};

const Map = ({ ISOCode, locale, coordinates }) => {
  const [map, setMap] = React.useState(null);

  const onLoad = React.useCallback(function callback(map) {
    map.data.loadGeoJson(
      `https://rock-paper-b93b9-default-rtdb.europe-west1.firebasedatabase.app/${ISOCode}.json`,
    );
    setMap(map);
  }, []);

  const onUnmount = React.useCallback(function callback(map) {
    setMap(null);
  }, []);

  const [lat, lng] = coordinates;
  const center = { lat, lng };

  const image = '/images/pin.png';

  const renderMap = () => (
    <LoadScript googleMapsApiKey={'AIzaSyCq9XpVQMtB0rLVqgjgWwQgcTJmDu5mRLc'} language={locale}>
      <GoogleMap
        mapContainerStyle={containerStyle}
        center={center}
        zoom={6}
        options={options}
        onLoad={onLoad}
        onUnmount={onUnmount}>
        <Marker position={center} icon={image} />
      </GoogleMap>
    </LoadScript>
  );

  return renderMap();
};

export default Map;
