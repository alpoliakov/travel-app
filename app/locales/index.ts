export const messages = {
  en: {
    btnLogin: 'Login',
    btnSignUp: 'Sign Up',
    search: 'Search',
    phEmail: 'Enter your e-mail',
    phPass: 'Enter your password',
    createAcc: 'Create an account',
    return: 'Return ?',
    enter: 'Enter',
    load: 'Loading',
    high: 'Max',
    low: 'Min',
    now: 'Currently',
    pass: 'Password (6 characters minimum):',
    phName: 'Enter your name',
    phAvatar: 'Upload your avatar',
    name: 'Name (3 characters minimum):',
    avatar: 'Avatar:',
    register: 'Sign Up',
    signOut: 'Sign Out',
    errorEmail: 'Invalid email field!',
    errorPass: 'Invalid password field!',
    errorName: 'Invalid username field!',
    profile: 'Profile',
    delete: 'Delete',
    or: 'Or',
    selectFile: 'Select a file',
    capital: 'Capital',
    hello: 'Hello!',
    modalText: 'To view, you need to log in or register!',
    close: 'Close',
    ratingQuestion: 'How do you rate this place?',
    opinion: 'Your opinion matters to us!',
    rateButton: 'Rate now',
    later: 'Maybe later',
    reminder: 'You need to login or register!',
    feedbackGive: 'give feedback',
    feedbackSee: 'see reviews',
    usd: 'USD',
    eur: 'EUR',
    rub: 'RUB',
  },
  ru: {
    btnLogin: 'Войти',
    btnSignUp: 'Регистрация',
    search: 'Поиск',
    phEmail: 'Введите ваш e-mail',
    phPass: 'Введите ваш пароль',
    createAcc: 'Создать аккаунт',
    return: 'Вернуться ?',
    enter: 'Ввести',
    load: 'Загрузка',
    high: 'Макс',
    low: 'Мин',
    now: 'Сейчас',
    pass: 'Пароль (6 символов минимум):',
    phName: 'Введите ваше имя',
    phAvatar: 'Загрузите ваш аватар',
    name: 'Имя (3 символа минимум):',
    avatar: 'Аватар:',
    register: 'Регистрация',
    signOut: 'Выйти',
    errorEmail: 'Невалидное поле электронной почты!',
    errorPass: 'Невалидное поле пароля!',
    errorName: 'Невалидное поле имени пользователя!',
    profile: 'Профиль',
    delete: 'Удалить',
    or: 'Или',
    selectFile: 'Выберите файл',
    capital: 'Столица',
    hello: 'Приветствуем!',
    modalText: 'Для просмотра необходимо авторизоваться или зарегистрироваться!',
    close: 'Закрыть',
    ratingQuestion: 'Как вы оцениваете это место?',
    opinion: 'Ваше мнение важно для нас!',
    rateButton: 'Оценить сейчас',
    later: 'Может быть, позже',
    reminder: 'Вам необходимо войти или зарегистрироваться!',
    feedbackGive: 'оставить отзыв',
    feedbackSee: 'посмотреть отзывы',
    usd: 'Доллар',
    eur: 'Евро',
    rub: 'Рубль',
  },
  uk: {
    btnLogin: 'Увійти',
    btnSignUp: 'Реєстрація',
    search: 'Пошук',
    phEmail: 'Введіть ваш e-mail',
    phPass: 'Введіть ваш пароль',
    createAcc: 'Створити акаунт',
    return: 'Повернутись ?',
    enter: 'Ввести',
    load: 'Завантаження',
    high: 'Макс',
    low: 'Miн',
    now: 'Зараз',
    pass: 'Пароль (6 символів мінімум):',
    phName: 'Введіть ваше им`я',
    phAvatar: 'Завантажте ваш аватар',
    name: 'Ім`я (3 символа мінімум):',
    avatar: 'Аватар:',
    register: 'Реєстрація',
    signOut: 'Вийти',
    errorEmail: 'Невалідне поле електронної пошти!',
    errorPass: 'Невалідне поле пароля!',
    errorName: 'Невалідне поле імені користувача',
    profile: 'Профіль',
    delete: 'Видалити',
    or: 'Або',
    selectFile: 'Виберіть файл',
    capital: 'Столиця',
    hello: 'Вітаємо!',
    modalText: 'Для перегляду потрібно увійти або зареєструватися!',
    close: 'Закрити',
    ratingQuestion: 'Як ви оцінюєте це місце?',
    opinion: 'Ваша думка для нас важлива!',
    rateButton: 'Оцініть зараз',
    later: 'Можливо пізніше',
    reminder: 'Вам необхідно увійти або зареєструватися!',
    feedbackGive: 'залишити відгук',
    feedbackSee: 'подивитися відгуки',
    usd: 'Долар',
    eur: 'Євро',
    rub: 'Рубль',
  },
};
