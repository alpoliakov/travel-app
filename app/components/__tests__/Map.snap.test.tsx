import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';

import { messages } from '../../locales';
import Map from '../Map';

const coordinates = [52.31, 13.23];
const ISOCode = 'DEU';

it('Map renders correctly', () => {
  const tree = renderer
    .create(
      <IntlProvider locale="ru" messages={messages['ru']}>
        <Map coordinates={coordinates} ISOCode={ISOCode} locale="ru" />)
      </IntlProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
