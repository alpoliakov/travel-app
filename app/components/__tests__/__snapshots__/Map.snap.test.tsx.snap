// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders correctly 1`] = `
Array [
  <div />,
  <div>
    Loading...
  </div>,
  ")",
]
`;
